.github/workflows/makefile-lint.yml

```
@@ -0,0 +1,96 @@
+            printf "%s\n" "No Makefiles found."
+            printf "%s\n" "files=" >> "$GITHUB_OUTPUT"
+          else
+            printf "%s\n" "Makefiles found:"
+            printf "%s\n" "$FILES"
+            # Replace line breaks with spaces for GitHub Action Output
+            FILES="${FILES//$'\n'/ }"
+            printf "%s\n" "files=$FILES" >> "$GITHUB_OUTPUT"
+          fi
+        if: ${{ success() }}
+      - name: Lint Makefiles Files
+        run: |
```

actionlint

shellcheck reported issue in this script: SC1009:info:2:27: The mentioned syntax error was in this then clause

(shellcheck)